File: vps-renew-notification/src/payload/provider.rs
use serde::Deserialize;
use validator::Validate;

#[derive(Deserialize, Validate)]
pub struct AddProvider {
    #[validate(length(min = 1, max = 50, message = "请输入正确的名称"))]
    pub name: String,

    #[validate(range(min = 1, message = "请输入正确的续期时间"))]
    pub renew_days: i32,

    #[validate(range(min = 1, message = "请输入正确的通知时间"))]
    pub notify_days: i32,
}

#[derive(Deserialize, Validate)]
pub struct EditProvider {
    #[validate(length(min = 20, max = 20, message = "请输入正确的ID"))]
    pub id: String,

    #[validate(length(min = 1, max = 50, message = "请输入正确的名称"))]
    pub name: String,

    #[validate(range(min = 1, message = "请输入正确的续期时间"))]
    pub renew_days: i32,

    #[validate(range(min = 1, message = "请输入正确的通知时间"))]
    pub notify_days: i32,
}

#[derive(Deserialize)]
pub enum ListProviderSort {
    #[serde(rename = "name")]
    Name,

    #[serde(rename = "name_desc")]
    NameDesc,

    #[serde(rename = "id")]
    ID,

    #[serde(rename = "id_desc")]
    IDDesc,
}

#[derive(Deserialize)]
pub struct ListProvider {
    pub name: Option<String>,
    pub sort: Option<ListProviderSort>,
}

File: vps-renew-notification-go/web/payload/provider.go
package payload

type AddProvider struct {
	Name       string `json:"name" form:"name" binding:"required,max=50"`
	RenewDays  int    `json:"renew_days" form:"renew_days" binding:"required,gte=1"`
	NotifyDays int    `json:"notify_days" form:"notify_days" binding:"required,gte=1"`
}
type EditProvider struct {
	ID string `json:"id" form:"id" binding:"required,len=20"`
	AddProvider
}

type ListProviderSort = string

const (
	ListProviderSortName     ListProviderSort = "name"
	ListProviderSortNameDesc ListProviderSort = "name_desc"
	ListProviderSortID       ListProviderSort = "id"
	ListProviderSortIDDesc   ListProviderSort = "id_desc"
)

type ListProvider struct {
	Name string           `json:"name" form:"name"`
	Sort ListProviderSort `json:"sort" form:"sort"`
}
